lib/queue.js

```diff
@@ -603,14 +642,34 @@ Queue.prototype.startMoveUnlockedJobsToWait = function() {
   }
 };
 
-Queue.prototype.processJobs = function(resolve, reject){
-  var _this = this;
-  var processJobs = this.processJobs.bind(this, resolve, reject);
 
+Queue.prototype.processJobs = function(resolve, reject, options){
+  var _this = this;
+  var processJobs = this.processJobs.bind(this, resolve, reject, options);
   if(!this.closing){
+    var idleTime = 0;
+    var obeyRateLimit = function() {};
```

Comment:
same here as in run, refactor outside processJobs. These two methods are quite critical so better to keep them as small and clean as possible.